test/images/agnhost/resource-consumer-controller/controller.go

```diff
@@ -43,8 +46,31 @@ var (
 	consumerPort             int
 	consumerServiceName      string
 	consumerServiceNamespace string
+	dnsDomain                string
 )
 
+func getDNSDomain() string {
+	if dnsDomain != "" {
+		return dnsDomain
+	}
+	dnsSuffixList, err := exec.Command("/agnhost", "dns-suffix").Output()
```

Comment:
This will actually break Windows support. You can't run binaries without any extension on Windows, so this automatically fails.

I don't know why you have to do an exec call. Why not just import and use ``agnhost/dns`` module directly?

Reply:
Docker image entrypoint for Linux and Windows is the same according to the [1] and [2]. So, looking at it I assume that either agnhost windows image is broken or this code must be working.

I don't have Windows cluster to test it there.

[1] https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/Dockerfile
[2] https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/Dockerfile_windows

Reply:
Docker works a bit differently, so having ``/agnhost`` as an entrypoint on a Windows image is fine. This is backed by the fact that we've been using this image in our CIs for a very long time.

With your changes, this is what it'll be printed out when using the image on a Windows node: ``panic: exec: "/agnhost": file does not exist``. I've already have to investigate this issue for the resource-consumer image.

Nonetheless, my point still stands: why not just import the ``agnhost/dns`` module and use it? It's faster, safer, and it makes more sense.

Reply:
I can reuse functions directly. Just keep in mind that it is "private" as of now and cannot be imported from another module. So, I need to make it public (update func name to start with Uppercase letter). Does it work for you?

Reply:
@claudiubelu, Ok, uploaded change with the direct func usage. Please, re-review.